tubular/capping.py

```diff
@@ -61,10 +63,10 @@ class CappingTransformer(BaseTransformer):
 
     def __init__(
         self,
-        capping_values=None,
-        quantiles=None,
-        weights_column=None,
-        **kwargs,
+        capping_values: dict[str, list[int | float | None]] | None = None,
+        quantiles: dict[str, list[int | float]] | None = None,
```

Comment:
I think that I'd disagree here; `quantiles` can _only_ be a dict of key-value pairs between 0 and 1 (see error block on line 89). 

I'd query why `int` is permitted at all given that only 0L and 1L are permissible values. In fact, [this link](https://stackoverflow.com/a/50928627) insists that we should be using `float` *only*, as `int` is implied.

> Rather than requiring that users write import numbers and then use numbers.Float etc., this PEP proposes a straightforward shortcut that is almost as effective: when an argument is annotated as having type float, an argument of type int is acceptable; similar, for an argument annotated as having type complex, arguments of type float or int are acceptable.

(from [PEP484 Type Hints](https://peps.python.org/pep-0484/#the-numeric-tower))

Reply:
Reading through the conversation there, it seems as though people still seem to write `float | int` in the wild. I'd rather we left this as it is.

Reply:
Looking at [OutOfRangeNullTransformer](https://github.com/lvgig/tubular/blob/fab26402081627f6f5f71bf26f3b84b4f1a2fc18/tubular/capping.py#L406):

```python
class OutOfRangeNullTransformer(CappingTransformer):
    ...
    def __init__(
        self,
        capping_values: dict[str, list[int | float | None]] | None = None,
        quantiles: dict[str, list[int | float | None]] | None = None,
        weights_column: str | None = None,
        **kwargs: dict[str, bool],
    ) -> None:
        super().__init__(
            capping_values=capping_values,
            quantiles=quantiles,
            weights_column=weights_column,
            **kwargs,
        )
```
You're right, there is a type hint for `None` to `quantiles` in there. But I think this a mistake as it gets passed to the parent class constructor, [CappingTransformer](https://github.com/lvgig/tubular/blob/fab26402081627f6f5f71bf26f3b84b4f1a2fc18/tubular/capping.py#L14):

```python
class CappingTransformer(BaseTransformer):
   ...
    def __init__(
        self,
        capping_values: dict[str, list[int | float | None]] | None = None,
        quantiles: dict[str, list[int | float]] | None = None,
        weights_column: str | None = None,
        **kwargs: dict[str, bool],
    ) -> None:

```

Which does not permit `None`. This is an error - will change.